documentation/library_guide/parallel_api/parallel_range_algorithms.rst

```diff
@@ -0,0 +1,104 @@
+* A range adaptor transforms its underlying data range(s) or view(s) into a new view with modified behavior.
+
+The following C++ standard random access adaptors and factories can be used with the |onedpl_short|
+parallel range algorithms:
+
+* ``std::ranges::views::all``: A range adaptor that returns a view that includes all elements of a range.
+* ``std::ranges::subrange``: A utility that produces a view from an iterator and a sentinel or from a range.
+* ``std::ranges::iota_view``: A range factory that generates a sequence of elements by repeatedly incrementing
+  an initial value.
+* ``std::ranges::single_view``: A view that contains exactly one element of a specified value.
+* ``std::ranges::transform_view``: A range adaptor that produces a view that applies a transformation to each element
+  of another view.
+* ``std::ranges::reverse_view``: A range adaptor that produces a reversed sequence of elements provided by another view.
+* ``std::ranges::take_view``: A range adaptor that produces a view of the first N elements from another view.
+* ``std::ranges::drop_view``: A range adaptor that produces a view excluding the first N elements from another view.
+
+Supported Algorithms
+--------------------
+
+The ``<oneapi/dpl/algorithm>`` header defines the parallel range algorithms in the ``namespace oneapi::dpl::ranges``.
+All algorithms work with both standard aligned (host) and device execution policies.
+
+The ``ONEDPL_HAS_RANGE_ALGORITHMS`` :ref:`feature macro <feature-macros>` may be used to test for presence of
+parallel range algorithms.
+
+.. _range-algorithms-202409L:
+
+If ``ONEDPL_HAS_RANGE_ALGORITHMS`` is defined to ``202409L`` or a greater value, the following algorithms are provided:
+
+* ``for_each``
+* ``transform``
+* ``find``
+* ``find_if``
+* ``find_if_not``
+* ``adjacent_find``
+* ``all_of``
+* ``any_of``
+* ``none_of``
+* ``search``
+* ``search_n``
+* ``count``
+* ``count_if``
+* ``equal``
+* ``sort``
+* ``stable_sort``
+* ``is_sorted``
+* ``min_element``
+* ``max_element``
+* ``copy``
+* ``copy_if``
+* ``merge``
+
+Usage Example for Parallel Range Algorithms
+-------------------------------------------
+
+.. code:: cpp
+
+    {        
+        std::vector<int> vec_in = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        std::vector<int> vec_out(vec_in.size());
+
+        auto view_in = std::ranges::views::all(vec_in) | std::ranges::views::reverse;
+        oneapi::dpl::ranges::copy(oneapi::dpl::execution::par, view_in, vec_out);
+    }
+    {
+        using usm_shared_allocator = sycl::usm_allocator<int, sycl::usm::alloc::shared>;
+
+        std::vector<int, usm_shared_allocator> vec_in = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        std::vector<int, usm_shared_allocator> vec_out(vec_in.size());
+
+        auto view_in = std::ranges::subrange(vec_in.begin(), vec_in.end()) | std::ranges::views::reverse;
+        oneapi::dpl::ranges::copy(oneapi::dpl::execution::dpcpp_default, view_in, std::span(vec_out));
+    }
```

Comment:
I faced issues while trying to run the example. Below is a fixed version.

```suggestion
    {
        sycl::queue q{};
        using usm_shared_allocator_type = sycl::usm_allocator<int, sycl::usm::alloc::shared>;
        usm_shared_allocator_type alloc{q};
        std::vector<int, usm_shared_allocator_type> vec_in({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, alloc);
        std::vector<int, usm_shared_allocator_type> vec_out(vec_in.size(), alloc);
        auto view_in = std::ranges::subrange(vec_in.begin(), vec_in.end()) | std::ranges::views::reverse;
        oneapi::dpl::ranges::copy(oneapi::dpl::execution::make_device_policy(q), view_in, std::span(vec_out));
    }
```

The problem was that `sycl::usm_allocator` does not have a default constructor, hence the used std::vector constructors were invalid.

Reply:
Also fixed.